### PR TITLE
Derive `Clone` and implement `{url, client}` for `AsyncClient`

### PR DESCRIPTION
**Pull request by Hirrolot**
Summary:
 - `#[derive(Clone)]` for `AsyncClient` so that it can now be cloned. The implementation should be cheap because `reqwest::Client` is wrapped in `Arc` automatically, and the `url` field should not be large either.
 - Add the `{url, client}` getters. They are helpful when you want to obtain data from which `AsyncClient` was constructed and don't want to pass these parameters around your code.

### CODE REVIEW

**Review by tnull — COMMENTED** [review]
Generally looks good. 
For the sake of feature parity, could you include the same for the `blocking` side, i.e., add `fn url()` and `fn agent()` there?

**Comment by Hirrolot** [comment]
Done.

**Review by tnull — APPROVED** [review]

**Comment by Hirrolot** [comment]
FYI: GitHub requires a maintainer's approval for the CI workflows to be run.

**Comment by tnull** [comment]
> FYI: GitHub requires a maintainer's approval for the CI workflows to be run.

Right, I'm not a maintainer of this crate, just another contributor.

**Comment by vladimirfomene** [comment]
@Hirrolot, can you rebase these changes on the current master branch?

**Comment by Hirrolot** [comment]
Done.

**Review by vladimirfomene — APPROVED** [review]

**Review by vladimirfomene — APPROVED** [review]

**Comment by vladimirfomene** [comment]
@Hirrolot, please can you sign your commits and do a force-push

**Comment by vladimirfomene** [comment]
@Hirrolot, please can you sign your commits and do a force-push

**Comment by notmandatory** [comment]
Once this one is in it's time to make the release, see #32 :-)

**Comment by Hirrolot** [comment]
@vladimirfomene, done.

**Pull request merged**
